sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorClient/AnomalyDetectionLiveTests.cs

```diff
@@ -387,10 +387,19 @@ await foreach (MetricEnrichedSeriesData seriesData in client.GetMetricEnrichedSe
                 returnedKeys.Add(seriesData.SeriesKey);
             }
 
-            IEnumerable<List<KeyValuePair<string, string>>> expectedKvps = seriesKeys.Select(key => key.ToList());
-            IEnumerable<List<KeyValuePair<string, string>>> returnedKvps = returnedKeys.Select(key => key.ToList());
+            // Making sure count is exactly 2 because the logic below relies on that.
+            Assert.That(seriesKeys.Count, Is.EqualTo(2));
+            Assert.That(returnedKeys.Count, Is.EqualTo(2));
 
-            Assert.That(returnedKvps, Is.EquivalentTo(expectedKvps));
+            if (AreSame(seriesKeys[0], returnedKeys[0]))
+            {
+                Assert.That(seriesKeys[1], Is.EquivalentTo(returnedKeys[1]));
+            }
+            else
+            {
+                Assert.That(seriesKeys[0], Is.EquivalentTo(returnedKeys[1]));
+                Assert.That(seriesKeys[1], Is.EquivalentTo(returnedKeys[0]));
+            }
```

Comment:
Order of the keys returned changed after the update, and we found out our validation didn't really work as intended. The current approach is not very elegant, but it works.

Reply:
Why not sort before comparing?

Reply:
I tried sorting it with `OrderBy`, but it's a bit annoying to do it because a dimension key is an `IEnumerable` of kvps. We would need to write a short sorting anonymous function, I believe.